### PR TITLE
Added libpng

### PR DESCRIPTION
**Pull request by hkctkuy**

### CODE REVIEW

File: projects/libpng/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+libpng (Portable Network Graphics library) is a C Library for handling PNG images.
+
+## Build Docker
+
+    $ sudo docker build -t oss-sydr-fuzz-libpng .
+
+## Run Hybrid Fuzzing
+
+Unzip Sydr (`sydr.zip`) in `projects/zlib` directory:
```
Comment (SweetVishnya) [inline projects/libpng/README.md:11]: libpng

File: projects/libpng/build.sh [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+# 3. Build zlib alongside libpng
+################################################################################
+
+# Build targets for libfuzzer
+
+export CC=clang
+export CXX=clang++
+export CFLAGS="-g -fsanitize=fuzzer-no-link,address,integer,bounds,null,undefined,float-divide-by-zero"
+export CXXFLAGS="-g -fsanitize=fuzzer-no-link,address,integer,bounds,null,undefined,float-divide-by-zero"
+
```
Comment (SweetVishnya) [inline projects/libpng/build.sh:33]: Disable logging

https://github.com/glennrp/libpng/blob/libpng16/contrib/oss-fuzz/build.sh#L26

Comment (SweetVishnya) [reply projects/libpng/build.sh:33]: autoreconf -f -i

File: projects/libpng/build.sh [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+./configure
+make -j$(nproc) clean
+make -j$(nproc) all
+
+export CFLAGS="-g -fsanitize=fuzzer,address,integer,bounds,null,undefined,float-divide-by-zero"
+export CXXFLAGS="-g -fsanitize=fuzzer,address,integer,bounds,null,undefined,float-divide-by-zero"
+
+mkdir /corpus
+
+cp *.* /corpus
```
Comment (SweetVishnya) [inline projects/libpng/build.sh:43]: search for .png files

https://github.com/glennrp/libpng/blob/libpng16/contrib/oss-fuzz/build.sh#L46

File: projects/libpng/build.sh [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+make clean
+
+export CFLAGS="-g -fPIC"
+export CXXFLAGS="-g -fPIC"
+
+./configure
+make -j$(nproc) clean
+make -j$(nproc) all
+
+SRC="."
```
Comment (SweetVishnya) [inline projects/libpng/build.sh:64]: already defined

File: projects/libpng/build.sh [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+make clean
+
+export CFLAGS="-fprofile-instr-generate -fcoverage-mapping"
+export CXXFLAGS="-fprofile-instr-generate -fcoverage-mapping"
+
+./configure
+make -j$(nproc) clean
+make -j$(nproc) all
+
+SRC="."
```
Comment (SweetVishnya) [inline projects/libpng/build.sh:84]: already defined

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

**Comment by hkctkuy** [comment]
Update pull request

File: projects/libpng/libpng_read.toml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+#
+################################################################################
+
+[sydr]
+args = "-l debug -n 1024 --app-timeout 180"
+target = "/libpng_read_sydr @@"
+
+[libfuzzer]
+path = "/libpng_read_fuzzer"
+args = "/corpus"
```
Comment (SweetVishnya) [inline projects/libpng/libpng_read.toml:23]: add dict

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

**Pull request merged**